### DISCUSSION TITLE
undestanding complexified vectors (again)

### DISCUSSION THREAD

**Discussion by ftalbrecht** [pymor/pymor#1462] [Q&A] [answered]
In #1383, after #1408  has been introduced, I tried to adapt the dune-gdt bindings. Following the [required changes for the `FenicsVectorSpace`](https://github.com/pymor/pymor/pull/1408/files#diff-ac4598fe7ee3083402a4315c55c5224860896e09afc6f4d913639314acd633ddL129), I figured that [these changes would suffice](https://github.com/pymor/pymor/pull/1383/commits/5facb6abc74df88cda8726478735b0c17bec9bc8#diff-5153dc94a4bc6e59e5a702be6113212d36737650fe86260254ba2e76d2e0536aL118) for the `DuneXTVectorSpace`. However, I have the following strange behavior that I can not pin down: compare
```python
from pymor.bindings.dunegdt import DuneXTVectorSpace
print(DuneXTVectorSpace.vector_type)
print(DuneXTVectorSpace.real_vector_type)
```
which gives the desired
```python
pymor.bindings.dunegdt.ComplexifiedDuneXTVector
pymor.bindings.dunegdt.DuneXTVector
```
with
```python
from pymor.bindings.dunegdt import DuneXTVectorSpace
print(DuneXTVectorSpace(1).vector_type)
print(DuneXTVectorSpace(1).real_vector_type)
```
which gives
```python
dune.xt.la._la.IstlVector
pymor.bindings.dunegdt.DuneXTVector
```
which is plainly wrong.

Ideas, @sdrave @renefritze?

**Comment by renefritze** [comment, marked as answer]
> Ideas, @sdrave @renefritze?

Don't overwrite `vector_type` with the `__init__` arg of same name [here](https://github.com/pymor/pymor/pull/1383/files#diff-5153dc94a4bc6e59e5a702be6113212d36737650fe86260254ba2e76d2e0536aR121)?

**Comment by ftalbrecht** [comment]
On 15.11.21 15:28, René Fritze wrote:
> 
>> Ideas, @sdrave @renefritze?
> 
> Don't overwrite `vector_type` with the `__init__` arg of same name [here](https://github.com/pymor/pymor/pull/1383/files#diff-5153dc94a4bc6e59e5a702be6113212d36737650fe86260254ba2e76d2e0536aR121)?
> 
> 

Nice catch, thx!
